vagrant/base.sh

```diff
@@ -17,23 +17,33 @@
 
 set -e
 
-if [ -z `which javac` ]; then
-    apt-get -y update
-    apt-get install -y software-properties-common python-software-properties
-    add-apt-repository -y ppa:webupd8team/java
-    apt-get -y update
-
-    # Try to share cache. See Vagrantfile for details
-    mkdir -p /var/cache/oracle-jdk7-installer
-    if [ -e "/tmp/oracle-jdk7-installer-cache/" ]; then
-        find /tmp/oracle-jdk7-installer-cache/ -not -empty -exec cp '{}' /var/cache/oracle-jdk7-installer/ \;
+if [ $(cat /etc/system-release | grep -oE 'Fedora release [[:digit:]]+') == 'Fedora release 21' ]
+then #working with fedora 21
+    if [ -z `which javac` ]
```

Comment:
`which` doesn't seem to be installed on this base box... When testing, I converted this to `if [ ! -e "/usr/bin/javac" ]` and that made it work. Alternatively, to avoid hard coding the path to javac, you could install `which` before doing this test.